.circleci/config.yml

```diff
@@ -616,6 +624,10 @@ jobs:
 workflows:
   build-test-deploy:
     jobs:
+      # Running two or more build-test-deploy workflows concurrently may result in failures to
+      #  puppeteer-test, api-deploy-to-test or/and ui-deploy-to-test jobs.
+      - wait_until_previous_workflow_done:
+          <<: *filter-master-branch
```

Comment:
Adding a new job that waits for previously submitted `build-test-deploy` to finish. `ui-deploy-to-test`, `api-deploy-to-test` and `puppeteer-test` jobs requires it.